### PR TITLE
Add check a response's validity before adding it

### PR DESCRIPTION
**Pull request by benjimarshall**
Thanks to @kwunkiuma for spotting this bug!

### CODE REVIEW

File: DiscordBot.DataAccess/EventsSheetsService.cs [hunk 1]
```diff
@@ -520,7 +520,16 @@ private async Task<IEnumerable<EventResponse>> GetEventResponseOptionsAsync(int
             IEnumerable<EventResponse> responseColumns
         )
         {
-            var newRow = responseColumns
+            var responseList = responseColumns.ToList();
+
+            if (responseList.All(response => response.Emoji != responseEmoji))
```
Comment (FridaTveit) [inline DiscordBot.DataAccess/EventsSheetsService.cs:525]: Minor but I'd probably find `if (!responseList.Any(...))` a bit easier to understand? 🙂

File: DiscordBot.DataAccess/EventsSheetsService.cs [hunk 2]
```diff
@@ -548,6 +557,13 @@ IEnumerable<EventResponse> responseColumns
         {
             var index = responseColumns.ToList().FindIndex(responseOption => responseOption.Emoji == responseEmoji);
 
+            if (index == -1)
+            {
+                throw new ResponseNotFoundException(
+                    $"Response {responseEmoji} is not recognised for event {eventKey}"
+                );
+            }
```
Comment (FridaTveit) [inline DiscordBot.DataAccess/EventsSheetsService.cs:565]: Does this check really need to be duplicated in both add response methods? It looks like the logic is really the same for both so could be done for both of them in the calling method? What do you think? 🙂

**Review by FridaTveit — COMMENTED** [review]
A couple of comments, otherwise looks good 🙂

File: DiscordBot.DataAccess/EventsSheetsService.cs (hunk 1 above)
Comment (benjimarshall) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: Yeah sure! Changed in 1dfe1d6 🙂

File: DiscordBot.DataAccess/EventsSheetsService.cs (hunk 2 above)
Comment (benjimarshall) [reply DiscordBot.DataAccess/EventsSheetsService.cs]: Yeah I agree! Changed in 32ae737 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**